tests/tracer/telemetry/test_writer.py

```diff
@@ -0,0 +1,226 @@
+    assert telemetry_writer._sequence == 2
+
+    headers = httpretty.last_request().headers
+    assert "DD-Telemetry-Request-Type" in headers
+    assert headers["DD-Telemetry-Request-Type"] == "app-closed"
+
+
+def test_add_event_disabled_writer(telemetry_writer_disabled):
+    """asserts that add_event() does not create a telemetry request telemetry writer is disabled"""
+    payload = {"test": "123"}
+    payload_type = "test-event"
+
+    # Ensure events are NOT queued when telemetry is disabled
+    telemetry_writer_disabled.add_event(payload, payload_type)
+    assert len(telemetry_writer_disabled._events_queue) == 0
+    assert telemetry_writer_disabled._sequence == 1
+
+
+def test_add_integration(telemetry_writer):
+    """ensure integrations are queued when telemetry is enabled"""
+    telemetry_writer.add_integration("integration-t", True)
+    telemetry_writer.add_integration("integration-f", False)
+
+    assert telemetry_writer._integrations_queue == [
+        {
+            "name": "integration-t",
+            "version": "",
+            "enabled": True,
+            "auto_enabled": True,
+            "compatible": "",
+            "error": "",
+        },
+        {
+            "name": "integration-f",
+            "version": "",
+            "enabled": True,
+            "auto_enabled": False,
+            "compatible": "",
+            "error": "",
+        },
+    ]
+
+
+def test_add_integration_disabled_writer(telemetry_writer_disabled):
+    """asserts that add_integration() does not queue an integration when telemetry is disabled"""
+    telemetry_writer_disabled.add_integration("integration-name", False)
+    assert len(telemetry_writer_disabled._integrations_queue) == 0
+
+
+@mock.patch("ddtrace.internal.telemetry.writer.monotonic")
+def test_sending_event(mock_monotonic, mock_send_request_200, telemetry_writer):
+    """asserts that the telemetry writer sends a valid telemetry header and payload"""
+    mock_monotonic.return_value = 1642544540
+
+    payload_type = "app-closed"
+    # Add event to the queue
+    telemetry_writer.add_event({}, payload_type)
+    # send request to the agent
+    telemetry_writer.periodic()
+
+    # Ensure request headers were set on the sent request
+    headers = httpretty.last_request().headers
+    assert headers["Content-type"] == "application/json"
+    assert headers["DD-Telemetry-Request-Type"] == payload_type
+    assert headers["DD-Telemetry-API-Version"] == "v1"
+
+    # Ensure json encoded request was sent to the agent proxy
+    assert httpretty.last_request().parsed_body == {
+        "tracer_time": 1642544540,
+        "runtime_id": get_runtime_id(),
+        "api_version": "v1",
+        "seq_id": 1,
+        "application": get_application(config.service, config.version, config.env),
+        "host": get_host_info(),
+        "payload": {},
+        "request_type": payload_type,
+    }
+
+
+def test_periodic(mock_send_request_200, telemetry_writer):
```

Comment:
This test should validate that we make the correct http calls.

Reply:
Yup I will add assertions to make sure the correct number of requests are sent. We test the request headers and the request body of app-started and app-integrations-change in other tests so I won't duplicate that here